scarf/utils.py

```diff
@@ -249,3 +250,31 @@ def rolling_window(a, w):
         for j in range(m):
             b[i, j] = a[x:y, j].mean()
     return b
+
+
+def permute_in_chunks(size: int, chunk_size: int, seed: int = 42) -> List[np.ndarray]:
+    """
+    Permutate an array in chunks of a given size.
+    Args:
+        size: The size of the array to be permuted
+        chunk_size: The size of the chunks to permute
+    Returns:
+        A permuted array of the given size
+    Examples:
+    >>> permute_in_chunks(10, 3)
+    [array([2, 1, 0]), array([5, 4, 3]), array([8, 7, 6]), array([9])]
+    """
+    np.random.seed(seed)
+    arr = np.arange(size)
```

Comment:
If you permute before chunking, you don't have to permute once per chunk

also numpy seems to want us to use a generator; not sure if that is involved or not; just FYI.
![image](https://github.com/user-attachments/assets/b257169a-4ce7-4114-9f9d-ae6144aec197)